### DISCUSSION TITLE
get smoothed_x from differentiation

### DISCUSSION THREAD

**Discussion by DongweiYe** [dynamicslab/pysindy#490] [Q&A]
Hi all,

I am new to the SINDy and I am trying to use the smoothing functions in SINDy to deal with the noise data for parameter inference. I am currently using Kalman filter or Spline, e.g. 
`sfd = ps.SINDyDerivative(kind="kalman", alpha=0.1)`
`d_hat = sfd._differentiate(X, t)`

It offers me the derivative of state data directly. I want to also get access to the smoothed_x for other use. I do found there is a  `self.smoothed_x_.` in sindy_derivative.py, but I have no clue how to call it properly. Could you help me out? 

Thank you!
Dongwei

**Comment by Jacob-Stevens-Haas** [comment]
`BaseDifferentiation.smoothed_x_` is an attribute set to the smoothed x values from the last derivative calculation.  It isn't itself callable.  That weird API was introduced in order to maintain backwards compatibility, but I admit it's a hack.

Hopefully we provide a better differentiation API soon

**Reply by DongweiYe** [reply]
Thanks for the reply! I see. Do you have any alternative way to get smoothed x value?
